README.md

```diff
@@ -1,56 +1,71 @@
+master    | [![CircleCI](https://circleci.com/gh/cosmos/cosmos-sdk/tree/master.svg?style=shield)](https://circleci.com/gh/cosmos/cosmos-sdk/tree/master) | [![codecov](https://codecov.io/gh/cosmos/cosmos-sdk/branch/master/graph/badge.svg)](https://codecov.io/gh/cosmos/cosmos-sdk)
+
+**WARNING**: the libraries are still undergoing breaking changes as we get better ideas and start building out the Apps.
+
+**Note**: Requires [Go 1.9+](https://golang.org/dl/)
+
+
+## Overview
+
+The Cosmos SDK is a platform for building multi-asset Proof-of-Stake
+cryptocurrencies, like the [Cosmos Hub](https://cosmos.network). It is both a library for building applications and a set of tools for securely interacting with them.
+
+The goal of the Cosmos-SDK is to allow developers to easily create their custom interoperable blockchain applications within the Cosmos Network without having to code every single functionality of their application, thus abstracting away the complexities of building a Tendermint ABCI application. We envision the SDK as the `npm`-like framework to build secure blockchain applications on top of Tendermint.
+
+In terms of its design, the SDK optimizes flexibility and security. The framework is designed around a modular execution stack which allows applications to mix and match elements as desired. In addition, all modules are sandboxed for greater application security.
+
+It is based on two major principles:
+
+- **Composability**: Anyone can create a module for the Cosmos-SDK, and using the already-built modules in your blockchain is as simple as importing them into your application.  As a developer, you only have to create the modules required by your application that do not already exist.
+
+- **Capabilities**: The SDK is inspired by capabilities-based security, and informed by years of wrestling with blockchain state machines. Most developers will need to access other modules when building their own modules. Given that the Cosmos-SDK is an open framework, we designed it using object-capabilities (_ocaps_) based principles. This is because we assume that some modules are malicious. In practice, this means that instead of having each module keep an access control list to give access to other modules, each module implements `keepers` that can be passed to other modules to grant a pre-defined set of capabilities. For example, if an instance of module A's `keepers` is passed to module B, the latter will be able to call a restricted set of module A's functions. The capabilities of each `keeper` are defined by the module's developer, and it's the job of the application developer to instantiate and pass a `keeper` from module to module properly. For a deeper look at capabilities, you can read this [article](http://habitatchronicles.com/2017/05/what-are-capabilities/).

```

Comment:
> This is because we assume that some modules are malicious

`This is because` is a clumsy sentence transition from the previous sentence

Reply:
`This means that instead of having each module keep an access control list to give access to other modules`
-> 
`This means that instead of having each module keep an access control list for other modules`

Reply:
> job of the application developer to instantiate and pass a `keeper` from module to module properly

It's not their job to do it "properly" - that's a hand-wavey term here. It's their job to understand and audit the safety of foreign code from 3rd party modules based on the capabilities they are passing into each 3rd party module. <- let's change this sentence to reflect this more nuanced idea